maplibre/src/coords.rs

```diff
@@ -22,54 +22,50 @@ pub const EXTENT_UINT: u32 = 4096;
 pub const EXTENT_SINT: i32 = EXTENT_UINT as i32;
 pub const EXTENT: f64 = EXTENT_UINT as f64;
 pub const TILE_SIZE: f64 = 512.0;
-pub const MAX_ZOOM: usize = 32;
 
-// FIXME: MAX_ZOOM is 32, which means max bound is 2^32, which wouldn't fit in u32 or i32
-// Bounds are generated 0..=31
-pub const ZOOM_BOUNDS: [u32; MAX_ZOOM] = create_zoom_bounds::<MAX_ZOOM>();
+/// Maximal zoom level. Typically, the maximal zoom is 18 as described
+/// [here](https://wiki.openstreetmap.org/wiki/Slippy_map_tilenames#Zoom_levels).
+/// This implementation allows 30 because this covers all tile sizes that can be represented
+/// with signed 32-bit integers.
+pub(crate) const MAX_ZOOM: u8 = 30;
```

Comment:
This should probably be an associated constant of `ZoomLevel`, i.e. `ZoomLevel::MAX`.